ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java

```diff
@@ -0,0 +1,290 @@
+                          : GenesisFileModule.createGenesisModule(genesisFile)
+                      : GenesisFileModule.createGenesisModule(network))
+              .evmToolCommandOptionsModule(daggerOptions)
+              .metricsSystemModule(new PrometheusMetricsSystemModule())
+              .build();
+
+      final BlockHeader blockHeader =
+          BlockHeaderBuilder.create()
+              .parentHash(Hash.EMPTY)
+              .coinbase(Address.ZERO)
+              .difficulty(Difficulty.ONE)
+              .number(1)
+              .gasLimit(5000)
+              .timestamp(Instant.now().toEpochMilli())
+              .ommersHash(Hash.EMPTY_LIST_HASH)
+              .stateRoot(Hash.EMPTY_TRIE_HASH)
+              .transactionsRoot(Hash.EMPTY)
+              .receiptsRoot(Hash.EMPTY)
+              .logsBloom(LogsBloomFilter.empty())
+              .gasUsed(0)
+              .extraData(Bytes.EMPTY)
+              .mixHash(Hash.EMPTY)
+              .nonce(0)
+              .blockHeaderFunctions(new MainnetBlockHeaderFunctions())
+              .buildBlockHeader();
+
+      Configurator.setAllLevels("", repeat == 0 ? Level.INFO : Level.OFF);
+      do {
+        final MessageFrame messageFrame =
+            MessageFrame.builder()
+                .type(MessageFrame.Type.MESSAGE_CALL)
+                .messageFrameStack(new ArrayDeque<>())
+                .blockchain(component.getBlockchain())
+                .worldState(component.getWorldUpdater())
+                .initialGas(gas)
+                .contract(Address.ZERO)
+                .address(receiver)
+                .originator(sender)
+                .gasPrice(gasPriceGWei)
+                .inputData(callData)
+                .sender(Address.ZERO)
+                .value(ethValue)
+                .apparentValue(ethValue)
+                .code(new Code(codeHexString))
+                .blockHeader(blockHeader)
+                .depth(0)
+                .completer(c -> {})
+                .miningBeneficiary(blockHeader.getCoinbase())
+                .blockHashLookup(new BlockHashLookup(blockHeader, component.getBlockchain()))
+                .contractAccountVersion(Account.DEFAULT_VERSION)
+                .build();
+
+        messageFrame.setState(MessageFrame.State.CODE_EXECUTING);
+        final EVM evm = component.getEvmAtBlock().apply(0);
+
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        evm.runToHalt(
+            messageFrame,
+            (frame, currentGasCost, executeOperation) -> {
+              if (showJsonResults && repeat == 0) {
+                System.out.println(createEvmTraceOperation(messageFrame));
+              }
+              executeOperation.execute();
+            });
+        stopwatch.stop();
+
+        if (repeat == 0) {
+          System.out.println(
+              new JsonObject()
+                  .put(
+                      "gasUser",
```

Comment:
gasUser ? Typo ?

Reply:
This output is matching existing outputs from other clients.  My understanding is that it was used for cross-client fuzzing at one point and I want to maintain maximum compatibility.

Reply:
Ok.